### PR TITLE
Outline library interface and command structure

### PR DESCRIPTION
**Pull request by AidanDelaney**
Document and outline the library interface

Signed-off-by: Aidan Delaney <adelaney21@bloomberg.net>

### CODE REVIEW

**Comment by AidanDelaney** [comment]
If/when this PR lands I'll introduce a representation of a template.  Then we will build up the rest of the library.

File: README.md [hunk 1]
```diff
@@ -1,2 +1,129 @@
...
+[![GitHub license](https://img.shields.io/github/license/buildpacks/scafall)](https://github.com/buildpacks/scafall/blob/main/LICENSE)
+[![Slack](https://img.shields.io/badge/slack-join-ff69b4.svg?logo=slack)](https://slack.cncf.io/)
+[![Gitpod ready-to-code](https://img.shields.io/badge/Gitpod-ready--to--code-blue?logo=gitpod)](https://gitpod.io/#https://github.com/buildpacks/scafall)
+
+
+A project scaffolding tool inspired by [cookiecutter](https://github.com/cookiecutter/cookiecutter).
+
+## Problem
+
+We needed a tool to create new source code projects from templates.  In addition, we needed the tool to be a libaray written in [Go](https://go.dev/).  Scafall takes project templates, asks the end-user some questions and produces an output folder.
```
Comment (natalieparellano) [inline README.md:16]:
```suggestion
We needed a tool to create new source code projects from templates.  In addition, we needed the tool to be a library written in [Go](https://go.dev/).  Scafall takes project templates, asks the end-user some questions and produces an output folder.
```

**Comment by natalieparellano** [comment]
RFC: https://github.com/buildpacks/rfcs/blob/main/text/0107-pack-buildpack-new-alternatives.md

File: pkg/scafall.go [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+// Scaffold accepts url containing project templates and creates an output
+// project.  The url can either point to a project template or a collection of
+// project templates.
+func (s Scafall) Scaffold() error {
+	return nil
+}
+
+// TemplateArguments returns a list of variable names that can be passed to the template
+func (s Scafall) TemplateArguments() (string, []string, error) {
```
Comment (hone) [inline pkg/scafall.go:67]: for completeness, would love to see a test in `pkg/example_scafall_test.go`, but also understand this just an interface PR.

File: pkg/scafall.go [hunk 3]
```diff
@@ -0,0 +1,69 @@
+// Scafall creates new source projects from project templates.  Project
+// templates are stored in git repositories and new source projects are created
+// on your local filesystem.
+package scafall
+
+// Scafall allows programmatic control over the default values for variables.
+// Any provided Arguments cause prompts for the same variable name to be skipped.
+type Scafall struct {
+	URL          string
```
Comment (hone) [inline pkg/scafall.go:9]: I know this isn't clarified in the RFC, but just thinking would URL be any valid `git` URL, i.e. can I pass in a local path as well for local dev of a template? Maybe I would use this/`pack` for tests of my template if I'm building a libcnb library.

File: pkg/scafall.go (hunk 2 above)
Comment (AidanDelaney) [reply pkg/scafall.go:67]: That's coming in the next PR.

**Pull request merged**